sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Partition.java

```diff
@@ -85,7 +141,14 @@
    * @throws IllegalArgumentException if {@code numPartitions <= 0}
    */
   public static <T> Partition<T> of(int numPartitions, PartitionFn<? super T> partitionFn) {
-    return new Partition<>(new PartitionDoFn<T>(numPartitions, partitionFn));
+
+    Contextful ctfFn =
+        Contextful.fn(
+            (T element, Contextful.Fn.Context c) ->
+                partitionFn.partitionFor(element, numPartitions),
+            Requirements.empty());
+    Object aClass = partitionFn;
```

Comment:
This is a no-op, please remove.

Reply:
Hi, I don't get your suggestion here. I will need to wrap interface function in ContextFul.Fn. Can you eloborate please?

Reply:
The statement `Object aClass = partitionFn;` has no effect. You can just pass `partitionFn` directly into the function.